server/src/domains/roles/roles.guard.ts

```diff
@@ -27,30 +36,7 @@ export class RolesGuard implements CanActivate {
         return true;
       }
 
-      const req = context.switchToHttp().getRequest();
-
-      const authorizationHeader = req.headers.authorization;
-
-      if (!authorizationHeader) {
-        throw new UnauthorizedException();
-      }
-
-      const bearer = authorizationHeader.split(' ')[0];
-      const accessToken = authorizationHeader.split(' ')[1];
-
-      if (bearer !== 'Bearer' || !accessToken) {
-        throw new UnauthorizedException();
-      }
-
-      const userPayload = this.authService.getUserPayloadFromToken(accessToken);
-
-      if (!userPayload) {
-        throw new UnauthorizedException();
-      }
-
-      req.user = userPayload;
-
-      const isAccess = userPayload.roles.some((role) =>
+      const isAccess = user.roles.some((role) =>
```

Comment:
What if required roles are:

- ADMIN
- ACCOUNTANT


But user is only ADMIN.

Example: a page with financial data, where only the admin can get and only the accountant can do something with the money on this page.